### PR TITLE
Dynamically generate the login screen

### LINKED ISSUE #1273: replace motd to match OH2 or OH3
on login to the box

make sure to change motd when migrating OH2<->OH3

### LINKED ISSUE #1334: Openhabian creates .bash_profile that does not include .profile
## Issue information:
openhabian creates / overwrites a .bash_profile which does not load .profile or does not take earlier changes into account.
Including .profile needs to be added manually to .bash_profile.
This can cause issues when other software than openhab is running on the machine.

## System information:
cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian

uname -m
armv7l

### PR DESCRIPTION
**Pull request by ecdye**
Fixes #1273
Fixes #1334

Signed-off-by: Ethan Dye <mrtops03@gmail.com>

### CODE REVIEW

**Comment by ecdye** [comment]
@mstormi Will this work? I think it will be much simpler and save me a lot a coding.

**Comment by mstormi** [comment]
No I don't think so. If someone changes branch but does un/reinstall OH it will not.
Also .bash_profile should not be system-manager, users should be able to edit it.

> and save me a lot a coding.

now that explains your reluctance but I don't understand why you think it is much coding ? what about [my proposal ](https://github.com/openhab/openhabian/issues/1273#issuecomment-749805493) ? That's just the extended version of what you did here, isn't it.

**Comment by ecdye** [comment]
The issue is that the banner is generated with certain components that are sourced from dpkg and others.

I dont see how I can circumvent that sourcing of version components without some very hacky coding.

**Comment by mstormi** [comment]
You mean this part?
```
OHVERSION=$(dpkg-query --showformat='${Version}' --show openhab2)
OHBUILD=$(sed -n 's/build-no\s*: //p' /var/lib/openhab2/etc/version.properties)
if [ "$OHBUILD" == "- release build -" ]; then OHBUILD="Release Build"; fi
```


source /opt/openhabian/functions/openhab.bash
if openhab3_is_installed; then OHPKG=openhab; else OHPKG=openhab2; fi

Then use $OHPKG instead of openhab2

or what else?

**Review by mstormi — CHANGES_REQUESTED** [review]
comments see last post

**Comment by ecdye** [comment]
> is there anything else that you believe you need hacky coding for to be circumvented?

Yeah how would I get the banner to take that input, the substitution of the variables happens in the banner itself and IIRC `cat` wont make the variable substitutions.

**Comment by mstormi** [comment]
use sed like we do in some locations (e.g. Amanda) to replace strings in templates in the includes/ dir

**Comment by ecdye** [comment]
Sounds good, gotta love when you just hit a wall. I'll work on this in the coming days. I won't have time to finish it tonight.

**Comment by mstormi** [comment]
FWIW I just discovered that openhabian.conf replacement didn't work in some cases on my own machine
Right after merging the PR, of course.
So quickly manually commented that out for now in both, master and openHAB3

**Comment by mstormi** [comment]
> FWIW I just discovered that openhabian.conf replacement didn't work in some cases

found, fixed & uploaded

**Comment by mstormi** [comment]
any progress, or did you hit another wall ?

**Comment by ecdye** [comment]
Haven't had time to start yet, but this is this weeks main project for me.

File: includes/bash_profile [hunk 1]
```diff
@@ -1,25 +1,22 @@
 # ~/.bash_profile: executed by bash(1) for interactive shells.
 
+source /opt/openhabian/functions/openhab.bash
+
 if [ -f "/opt/openHABian-install-failed" ]; then
-  echo ""
-  echo -e -n "\e[31;01mAttention! \e[39;49;00m"
+  echo -e -n "\\n\e[31;01mAttention! \e[39;49;00m"
```
Comment (mstormi) [inline includes/bash_profile:6]: see https://github.com/openhab/openhabian/pull/1392#discussion_r553246870

**Review by mstormi — CHANGES_REQUESTED** [review]
Just change the text to be more useful for users then we should be done and can merge

File: includes/bash_profile (hunk 1 above)
Comment (ecdye) [reply includes/bash_profile:6]: Will do when I have time.

**Review by mstormi — APPROVED** [review]

**Pull request merged**